R/Output_overview.R

```diff
@@ -0,0 +1,268 @@
+
+present_output <- function(OUTPUT) {
+
+  N_spec = 3
+
+  mosq_species = c("faruati", "punctulatus", "koliensis")
+
+  mosq_comp = c("EL_M_far", "LL_M_far", "P_M_far", "S_M_far", "E_M_far", "I_M_far",
+        "EL_M_pun", "LL_M_pun", "P_M_pun", "S_M_pun", "E_M_pun", "I_M_pun",
+                "EL_M_kol", "LL_M_kol", "P_M_kol", "S_M_kol", "E_M_kol", "I_M_kol")
+
+  mosq_comp = mosq_comp[1:(N_spec*6)]
+
+
+
+  colnames(OUTPUT) <- c("time",
+                        "S", "I_PCR", "I_LM", "D", "T", "P",
+                        mosq_comp,
+                        "N_pop", "PvPR_PCR", "PvPR_LM", "Pv_clin", "PvHR",
+                        "PvHR_batch", "new_PCR", "new_LM", "new_D", "new_T",
+                        "N_pop_U5", "PvPR_PCR_U5", "PvPR_LM_U5", "Pv_clin_U5", "PvHR_U5",
+                        "PvHR_batch_U5", "new_PCR_U5", "new_LM_U5", "new_D_U5", "new_T_U5",
+                        "N_pop_U10", "PvPR_PCR_U10", "PvPR_LM_U10", "Pv_clin_U10", "PvHR_U10",
+                        "PvHR_batch_U10", "new_PCR_U10", "new_LM_U10", "new_D_U10", "new_T_U10",
+            "EIR", "LLIN_cov", "IRS_cov", "ACT_treat", "PQ_treat", "pregnant", "A_par", "A_clin")
+
+  par(ask=TRUE)

```

Comment:
This plotting I think should be separated from the processing, and an interactively-stepped through series of plots seems like a suboptimal way of having to review things (the effect of `par(ask=TRUE)` is to request the user to step through each time a new plot is generated, in an interactive session).

Perhaps a simpler way to do this would be to translate this set of diagnostics into a small knitr report, store the Rmd for that into `inst/diagostics.Rmd` or similar, then compile a small report so that the effect of this function is to write out a html file - this could optionally displayed to the user in an interactive session too to capture more of the original intent.

In general though, functions should try not to alter the global state of the session, and calls to `par` that are not reset will do this.

@pwinskill probably has other ideas how this would fit into the malaria group workflows

Reply:
For our GTS use case we will mostly be aggregating up many runs before we summarise and visualise so I would not envisage much use for these built in plotting functions. 

I like the idea of the option for a automated diagnostic report to be produced if requested. I think this could be fairly simple (and might piggyback on some of the plotting functionality here). For example something like _Plot1: incidence over time, Plot2: prevalence over time, Plot3: prevalence~incidence and Plot4: intervention coverage over time_ would probably suffice to help us to spot most issues in individual simulations

Reply:
Ok, it seems like this diagnostic code is not immediately useful so I've added a [issue ](https://github.com/mrc-ide/vivax/issues/5) in this milestone to remove it.

But some different diagnostics could potentially be useful, so I've added a [separate issue](https://github.com/mrc-ide/vivax/issues/6) which we can prioritise if the model starts bugging out.